fog/load_testing/src/bin/ingest.rs

```diff
@@ -113,12 +114,44 @@ impl core::fmt::Display for TestResult {
     }
 }
 
+pub struct AutoKillChild(pub std::process::Child);
+
+impl AutoKillChild {
+    pub fn assert_not_stopped(&mut self) {
+        match self.0.try_wait() {
+            Ok(Some(stat)) => {
+                panic!("child stopped unexpectedly: status: {}", stat)
+            }
+            Ok(None) => {}
+            Err(err) => {
+                panic!("error getting child status: {}", err)
+            }
+        }
+    }
+}
+
+impl Drop for AutoKillChild {
```

Comment:
this improves the way the test works, because if the load testing binary panics (e.g. the test timed out), rust will kill this destructor and kill the child, so it doesn't spam your console and force you to `killall ...`

Reply:
compare with `std::process::Child`, which per documentation, does not kill the child in the destructor: https://doc.rust-lang.org/std/process/struct.Child.html